src/components/VaultItemTiles.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import {
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  Card,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+  Stack,
+  TextField,
+} from "@mui/material";
+import DeleteTwoToneIcon from "@mui/icons-material/DeleteTwoTone";
+
+export default function VaultItemTiles({ items }) {
+  const itemsArray = Object.keys(items).map((item) => items[item].data);
+  const [open, openValue] = useState(false);
+  const [vaultItem, setValue] = useState(0);
+
+  const openPopup = (event, value) => {
+    openValue(true);
+    setValue(value);
+  };
+
+  const closePopup = () => {
+    openValue(false);
+  };
+
+  return (
+    <>
+      <List>
+        {itemsArray.map((vaultItem) => (
+          <ListItem key={vaultItem.name}>
+            <Card sx={{ width: 500 }}>
+              <ListItemButton onClick={() => openPopup(event, vaultItem)}>
+                <ListItemText
+                  primary={vaultItem.name}
+                  secondary={vaultItem.username}
+                />
+                <DeleteTwoToneIcon />
+              </ListItemButton>
+            </Card>
+          </ListItem>
+        ))}
+      </List>
+      <Dialog open={open} fullWidth maxWidth="sm">
```

Comment:
I see what you mean. The reason I thought it would be used for updating is because the fields are input fields (mui's `TextField`) so users would be able to enter text. The next step would be wiring up the input fields to state variables and allow users to the update the vault items.

Do you think we will have time to work on Update? If not, to make this modal only display stuff and not accept user input, we can remove the input fields and use another element. 

A follow task to this display vault info modal is, I think the password should still be hidden and only be viewable if a user clicks on a button/icon to display the pw. I don't know if what we will get to this week, but I think for security purposes it's better to hide it.

Reply:
Okay cool! I think once the change is in, this PR will be in a good state.

Reply:
I ended up leaving them as textfields, just because the labelling looked the nicest and it was very easy to add the password masking. I did make them read only fields though so they are not editable. I can look into removing the blue focus when I am adding some styling if we do not like that.  I also moved it into its own component, so we can easily swap it with NewItemForm if we go that route. 

![ezgif com-video-to-gif (2)](https://github.com/secure-password-manager/PasswordManagerWeb/assets/67447055/90e389cd-ffd5-4c8d-b409-e903d7c4c77e)